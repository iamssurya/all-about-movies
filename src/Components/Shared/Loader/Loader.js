import React from "react";
import { Spinner } from "react-bootstrap";

const Loader = () => {
  return (
    <React.Fragment>
      <div>
        <div className="align-self-center">
          <Spinner animation="grow" variant="success" />
          <Spinner animation="grow" variant="danger" />
          <Spinner animation="grow" variant="warning" />
          <Spinner animation="grow" variant="info" />
          <Spinner animation="grow" variant="dark" />
        </div>
      </div>
    </React.Fragment>
  );
};

export default React.memo(Loader);
