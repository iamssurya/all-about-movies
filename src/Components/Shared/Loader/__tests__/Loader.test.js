import React from "react";
import Enzyme, { mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import Loader from "../Loader";

Enzyme.configure({ adapter: new Adapter() });

describe("<Loader /> component tests", () => {
  let Component;
  let Wrapper;
  beforeEach(() => {
    Component = mount(<Loader />);
    Wrapper = toJson(Component);
  });

  it("Should render component correctly", () => {
    expect(Wrapper).toMatchSnapshot();
  });
});
