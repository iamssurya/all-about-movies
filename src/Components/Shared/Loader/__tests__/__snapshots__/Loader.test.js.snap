// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> component tests Should render component correctly 1`] = `
<Memo(Loader)>
  <div
    className="loader"
  >
    <div
      className="align-self-center"
    >
      <Spinner
        animation="grow"
        variant="success"
      >
        <div
          className="spinner-grow text-success"
        />
      </Spinner>
      <Spinner
        animation="grow"
        variant="danger"
      >
        <div
          className="spinner-grow text-danger"
        />
      </Spinner>
      <Spinner
        animation="grow"
        variant="warning"
      >
        <div
          className="spinner-grow text-warning"
        />
      </Spinner>
      <Spinner
        animation="grow"
        variant="info"
      >
        <div
          className="spinner-grow text-info"
        />
      </Spinner>
      <Spinner
        animation="grow"
        variant="dark"
      >
        <div
          className="spinner-grow text-dark"
        />
      </Spinner>
    </div>
  </div>
</Memo(Loader)>
`;
